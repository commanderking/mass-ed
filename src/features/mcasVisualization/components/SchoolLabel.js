import React from 'react';

export const SchoolLabel = ({ schoolName, schoolIndex, deleteSchool }) => {
  return (
    <div className='schooLabelWrapper'>
      <span>{schoolName}</span>
      <button
        className="closeGraphButton"
        onClick={() => {
          deleteSchool(schoolIndex);
        }}
      >
        X
      </button>
    </div>
  )
};
