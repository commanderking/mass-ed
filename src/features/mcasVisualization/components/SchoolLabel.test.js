import React from "react";
import { shallow } from "enzyme";
import { SchoolLabel } from "./SchoolLabel";
import { Button } from "reactstrap";

describe("School Label", () => {
  let mockProps = {
    schoolName: "mock-school-name",
    schoolCode: 12345,
    deleteSchool: jest.fn(),
    index: 1
  };
  let wrapper = shallow(<SchoolLabel {...mockProps} />);

  it("renders correct structure", () => {
    expect(wrapper.find(".schoolLabelWrapper")).toHaveLength(1);
    expect(wrapper.find(".selectedSchoolsIcon")).toHaveLength(1);
    expect(wrapper.find(".selectedSchoolsIcon")).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it("button calls deleteSchool prop function", () => {
    const deleteSchoolButton = wrapper.find(Button);
    deleteSchoolButton.simulate("click");
    expect(mockProps.deleteSchool).toHaveBeenCalledTimes(1);
    expect(mockProps.deleteSchool).toHaveBeenCalledWith(mockProps.schoolCode);
  });
});
