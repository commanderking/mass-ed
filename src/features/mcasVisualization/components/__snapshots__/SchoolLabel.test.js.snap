// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`School Label renders correct structure 1`] = `
<div
  className="schoolLabelWrapper"
>
  <div
    className="selectedSchoolsIcon"
    style={
      Object {
        "backgroundColor": "#b19cd9",
      }
    }
  />
  <span>
    mock-school-name
  </span>
  <button
    className="closeGraphButton"
    onClick={[Function]}
  >
    X
  </button>
</div>
`;
